### PR TITLE
Fix textureWrap REPEAT for WebGL Power Of Two Image Element

### LINKED ISSUE #5668: Texture from IMG element does not use actual dimensions of image
### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [X] DOM
- [ ] Events
- [ ] Friendly error system
- [X] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Other (specify if possible)

### p5.js version

1.4.1

### Web browser and version

Version 1.36.112 Chromium: 99.0.4844.51 (Official Build) (x86_64)

### Operating System

MacOS 12.3.1

### Steps:
1. Use a file input to upload an image of power-of-two size (like 1024x1024) into an existing \ element that has a non-power-of-two size. Like 40x40.
2. Use that \ element as a texture with a texture wrap mode of REPEAT.
3. p5js warns that the image is not a power of two and that wrap mode will be changed to CLAMP.

Problem seems to be that it's using the width and height attributes of the \ element and should be using the naturalWidth and naturalHeight attributes.

### PR DESCRIPTION
**Pull request by kate-grant**
Resolves #5668
Previously `setWrapMode` checked `isPowerOfTwo` on element dimensions and not on src image dimensions. This caused `textureWrap(REPEAT)` to set to CLAMP if `this` was an HTML element with non-power-of-two element dimensions, but this outcome was designed for non-power-of-two src image dimensions. 

 Changes:

- I changed `p5.Texture.prototype.setWrapMode` to add `wrapWidth` and `wrapHeight` to differentiate between HTML element src image dimensions vs. the element dimensions.
- Added unit tests for textureWrap REPEAT on textures from image elements for power-of-two and non-power-of-two.
- The power-of-two case should allow REPEAT and the non-power-of-two case should set to CLAMP.

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
inline documentation already existed but the code did not perform as described
- [x] [Unit tests] are included / updated

### CODE REVIEW

File: test/unit/webgl/p5.Texture.js [hunk 1]
```diff
@@ -12,15 +15,27 @@ suite('p5.Texture', function() {
...
         //p.box(70, 70, 70);
       };
       p.setup = function() {
         canvas = p.createCanvas(100, 100, p.WEBGL);
         texImg1 = p.createGraphics(2, 2, p.WEBGL);
         p.texture(texImg1);
+        p.createImg(texImg2.canvas.toDataURL(), '', 'anonymous', el => {
+          el.size(50, 50);
+          imgElementPowerOfTwo = el;
+          p.texture(imgElementPowerOfTwo);
```
Comment (davepagurek) [inline test/unit/webgl/p5.Texture.js:32]: It looks like this is working right now, but maybe we can make it a tad clearer for readers of the tests which texture will be active at the end of `setup`? I think the success callback here will always run immediately due to it being a data URL, but I think `done()` gets called before those callbacks run. Maybe we could do something like this to ensure we always end with `texImg1` as the active texture, and that we call `done` after?

```js
new Promise(resolve => {
  p.createImg(texImg2.canvas.toDataURL(), '', 'anonymous', el => {
    el.size(50, 50);
    imgElementPowerOfTwo = el;
    p.texture(imgElementPowerOfTwo);
    resolve();
  });
}).then(() => new Promise(resolve => {
  p.createImg(texImg3.canvas.toDataURL(), '', 'anonymous', el => {
    el.size(50, 50);
    imgElementNotPowerOfTwo = el;
    p.texture(imgElementNotPowerOfTwo);
    resolve();
  });
})).then(() => {
  p.texture(texImg1);
  done();
});
```

**Review by davepagurek — COMMENTED** [review]
This solution looks great, and thanks for adding tests! I just have a comment about making the test setup a little more predictable but then I think this is good to go!

File: test/unit/webgl/p5.Texture.js (hunk 1 above)
Comment (kate-grant) [reply test/unit/webgl/p5.Texture.js]: sure thing--added this.

**Comment by davepagurek** [comment]
Awesome, thanks!

**Comment by kate-grant** [comment]
@davepagurek Thanks! Would it be ok to go ahead and add myself to the contributors list?

**Comment by davepagurek** [comment]
Oh yes thanks for reminding me!

**Comment by davepagurek** [comment]
@all-contributors please add @kate-grant for bug, code, test

**Comment by kate-grant** [comment]
@davepagurek thanks!

**Pull request merged**